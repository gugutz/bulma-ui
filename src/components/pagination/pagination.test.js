import React from 'react'
import renderer from 'react-test-renderer'
import Pagination from './pagination'

describe('<Pagination /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Pagination role="navigation" aria-label="pagination">
          <Pagination.Previous>Previous</Pagination.Previous>
          <Pagination.Next>Next page</Pagination.Next>
          <Pagination.List>
            <li>
              <Pagination.Link>1</Pagination.Link>
            </li>
            <li>
              <Pagination.Ellipsis>&hellip;</Pagination.Ellipsis>
            </li>
            <li>
              <Pagination.Link>45</Pagination.Link>
            </li>
            <li>
              <Pagination.Link isCurrent>46</Pagination.Link>
            </li>
            <li>
              <Pagination.Link>47</Pagination.Link>
            </li>
            <li>
              <Pagination.Ellipsis>&hellip;</Pagination.Ellipsis>
            </li>
            <li>
              <Pagination.Link>86</Pagination.Link>
            </li>
          </Pagination.List>
        </Pagination>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const positions = ['left', 'right', 'center']

  for (const position of positions) {
    it('should support position prop with ' + position, () => {
      const tree = renderer
        .create(
          <Pagination position={position}>
            <div>children</div>
          </Pagination>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support isRounded prop', () => {
    const tree = renderer
      .create(
        <Pagination isRounded>
          <div>children</div>
        </Pagination>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const sizes = ['small', 'medium', 'large', 'normal']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Pagination size={size}>
            <div>children</div>
          </Pagination>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
