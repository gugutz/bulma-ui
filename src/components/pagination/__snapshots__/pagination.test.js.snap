// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> component should render correctly 1`] = `
<nav
  aria-label="pagination"
  className="pagination"
  role="navigation"
>
  <a
    className="pagination-previous"
  >
    Previous
  </a>
  <a
    className="pagination-next"
  >
    Next page
  </a>
  <ul
    className="pagination-list"
  >
    <li>
      <a
        className="pagination-link"
      >
        1
      </a>
    </li>
    <li>
      <span
        className="pagination-ellipsis"
      >
        …
      </span>
    </li>
    <li>
      <a
        className="pagination-link"
      >
        45
      </a>
    </li>
    <li>
      <a
        className="pagination-link is-current"
      >
        46
      </a>
    </li>
    <li>
      <a
        className="pagination-link"
      >
        47
      </a>
    </li>
    <li>
      <span
        className="pagination-ellipsis"
      >
        …
      </span>
    </li>
    <li>
      <a
        className="pagination-link"
      >
        86
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> component should support isRounded prop 1`] = `
<nav
  className="pagination is-rounded"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support position prop with center 1`] = `
<nav
  className="pagination is-centered"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support position prop with left 1`] = `
<nav
  className="pagination"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support position prop with right 1`] = `
<nav
  className="pagination is-right"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support size prop with large 1`] = `
<nav
  className="pagination is-large"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support size prop with medium 1`] = `
<nav
  className="pagination is-medium"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support size prop with normal 1`] = `
<nav
  className="pagination"
>
  <div>
    children
  </div>
</nav>
`;

exports[`<Pagination /> component should support size prop with small 1`] = `
<nav
  className="pagination is-small"
>
  <div>
    children
  </div>
</nav>
`;
